### PR TITLE
Add ICN check

### PR DESCRIPTION
**Pull request by acovrig**
## Summary

- Require an ICN in v1 before connecting to MPI, returning a 422 if no ICN on the veteran.

## Related issue(s)
- [API-22592](https://vajira.max.gov/browse/API-22592)

## Testing done

- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected*
- *Describe the tests completed and the results*

## What areas of the site does it impact?
v1

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback
Does this need a test?

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v1/application_controller.rb [hunk 1]
```diff
@@ -59,9 +68,9 @@ def validate_veteran_identifiers(require_birls: false) # rubocop:disable Metrics
                               ptcpnt_id: target_veteran.participant_id.present?,
                               birls_id: target_veteran.birls_id.present?,
                               icn: target_veteran&.mpi_icn)
-      rescue ::Common::Exceptions::UnprocessableEntity, MPI::Errors::ArgumentError
+      rescue MPI::Errors::ArgumentError
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v1/application_controller.rb:71]: is this update from the MPI rework? Assuming we don't have the potential of the `UnprocessableEntity` anymore?

**Review by FonzMP — APPROVED** [review]
Pending the question above I'm fine with this. 

:shipit:

File: modules/claims_api/spec/requests/v1/claims_request_spec.rb [hunk 2]
```diff
@@ -164,6 +164,61 @@
...
+            vet.participant_id = nil
+            allow_any_instance_of(ClaimsApi::V1::ApplicationController)
+              .to receive(:veteran_from_headers).and_return(vet)
+
+            allow_any_instance_of(ClaimsApi::Veteran)
+              .to receive(:mpi_record?).and_return(false)
+
+            get '/services/claims/v1/claims/123123131', params: nil, headers: request_headers.merge(auth_header)
+
+            # expect(response.status).to eq(422)
```
[the remark this answers is not in the document]
Comment (acovrig) [reply modules/claims_api/spec/requests/v1/claims_request_spec.rb]: lol, not sure how I missed that, it's not exactly needed given the test below, but I re-enabled it anywho.

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**